okio-files/src/jsMain/kotlin/okio/FileSink.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import fs.writeSync
```

Comment:
This means you can't use it in any critical path on a server which is unfortunate.

Reply:
Agreed!

Reply:
We can discuss our suspending strategy here. I don’t think this PR precludes that.
https://github.com/square/okio/issues/814

Reply:
> This means you can't use it in any critical path on a server

Interesting, why so? Do we want something async instead?

Reply:
Node runs on a single thread just like browser JS. It's an event loop. Blocking I/O blocks the entire server. So it's fine for command-line utilities running on node but not a proper server.

Reply:
Gotcha, makes sense

Reply:
> Do we want something async instead?

The entire design is built around being synchronous. To wrap the forthcoming filesystem API in the browser and to support Node's regular async file APIs you basically need a parallel design.

Reply:
In particular, making this async makes it more difficult to use it in OkHttp’s cache, which is my initial target use case. OkHttp’s cache is blocking.